velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -363,7 +366,11 @@ int64_t parseTimezoneOffset(const char* cur, const char* end, Date& date) {
         if (std::strncmp(cur + 1, "00:00", 5) == 0) {
           date.timezoneId = 0;
         } else {
-          date.timezoneId = util::getTimeZoneID(std::string_view(cur, 6));
+          try {
+            date.timezoneId = util::getTimeZoneID(std::string_view(cur, 6));
+          } catch (VeloxRuntimeError& e) {
```

Comment:
There is limited value in failOnError flag if implementation still throws exceptions, but the catches them and converts to error result. The whole point is to avoid throwing exceptions when failOnError is false.

Reply:
@mbasmanova, yes, we can remove this try/catch. Just updated in a new commit. Thanks!